### ISSUE TITLE
GDK_PIXBUF_MODULE_FILE should not be set in DE modules (KDE, Xfce)

### ISSUE THREAD

**Issue by jaens** [NixOS/nixpkgs#33231]
## Issue description

Having both enabled at once in NixOS results in the following error:
`
error: The unique option 'environment.variables.GDK_PIXBUF_MODULE_FILE' is defined multiple times, in '/home/jaen/Repositories/nixpkgs/nixos/modules/services/x11/desktop-managers/plasma5.nix' and '/home/jaen/Repositories/nixpkgs/nixos/modules/services/x11/desktop-managers/xfce.nix'.
`

(on the `nixos-unstable` channel)

### Steps to reproduce

```
  services.xserver.desktopManager.plasma5.enable = true;
  services.xserver.desktopManager.xfce.enable = true;
```

## Technical details

```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.5, NixOS, 18.03.git.8989149 (Impala)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 1.11.16`
 - channels(root): `"nixos-18.03pre122666.afe9649210c"`
 - channels(jaen): `""`
 - nixpkgs: `/home/jaen/Repositories/nixpkgs`
```

**Comment by jaens** [comment]
Looks like the conflict was introduced in 209cbf9fc7e7d70d254abf457f154a934201f420
cc @yegortimoshenko

**Comment by yegortimoshenko** [comment]
Indeed.

**Comment by yegortimoshenko** [comment]
I'll make a superficial fix, but I'm really not sure how to really resolve this. Would it perhaps be a good idea to wrap all Gtk apps to have `GDK_PIXBUF_MODULE_FILE`?

/cc @jtojnar (GNOME), @ttuegel (KDE)

**Comment by yegortimoshenko** [comment]
@jaens: conflict has been fixed in https://github.com/NixOS/nixpkgs/commit/0dd6bd214da0b3ec79400a2910a0ca0a580eccf8, so you can enable KDE and Xfce at the same time, but underlying problem is not resolved yet, so I'm leaving the issue open.

**Comment by jtojnar** [comment]
IMHO, these things should be done in the derivation (for an example, see [nautilus](https://github.com/NixOS/nixpkgs/blob/0dd6bd214da0b3ec79400a2910a0ca0a580eccf8/pkgs/desktops/gnome-3/core/nautilus/default.nix#L20-L24)), possibly in some more automated way.

**Comment by ttuegel** [comment]
> IMHO, these things should be done in the derivation (for an example, see nautilus), possibly in some more automated way.

This is done by `wrapGAppsHook`. The problem is, KDE needs GTK applications to support SVG icons, which requires a different module file than the wrapper uses. It looks like Xfce needs the same thing.

The correct solution is for KDE and Xfce to make the setting in their respective startup scripts.

**Comment by yegortimoshenko** [comment]
As I understand, currently SVG icons in GTK apps are broken in most environments, the only reason it might not be entirely obvious is that most themes and apps also package raster icons. It's not Xfce's or KDE's fault in any way, and icons would be as broken in, say, just bare Xmonad.

I don't see any solution better than depending on librsvg in all GTK apps.

**Comment by burjui** [comment]
This is still a problem. Is there any solution?

**Comment by ambrop72** [comment]
I have developed a solution in pull request https://github.com/NixOS/nixpkgs/pull/42562. The change is against master but applies cleanly to 18.03. With this I can successfully enable Xfce and Plasma at the same time, and run Thunar from a Plasma session.

**Comment by jtojnar** [comment]
Fixed in #42562
